net/net.go

```diff
@@ -161,8 +161,21 @@ func BigForIP(ip net.IP) *big.Int {
 
 // AddIPOffset adds the provided integer offset to a base big.Int representing a
 // net.IP
-func AddIPOffset(base *big.Int, offset int) net.IP {
-	return net.IP(big.NewInt(0).Add(base, big.NewInt(int64(offset))).Bytes())
+func AddIPOffset(base *big.Int, offset int, isIPv6 bool) net.IP {
```

Comment:
do you know if modifying `BigForIP` to return the bigInt with correct length would solve the issue?
the benefit is that BigFortIP "knows" the right length and this way we don't have to modify this function passing an additional parameter

Reply:
Please don't change existing method signatures. If needed, deprecate this one and delegate to a second method that takes an ipv6 param

Reply:
modifying BigForIP is useless, because bigInt are normalized after the operations, we can add a new function that receives the IP as a parameters instead of the bigInt, I've tested this and it works

```
// AddNetIPOffset adds the provided integer offset to an IP address
func AddNetIPOffset(ip net.IP, offset int) net.IP {
	length := len(ip)
	base := new(big.Int).SetBytes(ip)
	ipBytes := big.NewInt(0).Add(base, big.NewInt(int64(offset))).Bytes()
	ipBytesLength := len(ipBytes)
	if ipBytesLength < length {
		b := make([]byte, length-ipBytesLength, length)
		return net.IP(append(b, ipBytes...))
	}
	return net.IP(ipBytes)
}
```